pydantic_ai_slim/pydantic_ai/models/grok.py

```diff
@@ -0,0 +1,625 @@
+"""Grok model implementation using xAI SDK."""
+
+import os
+from collections.abc import AsyncIterator, Iterator, Sequence
+from contextlib import asynccontextmanager
+from dataclasses import dataclass
+from typing import Any
+
+import xai_sdk.chat as chat_types
+
+# Import xai_sdk components
+from xai_sdk import AsyncClient
+from xai_sdk.chat import assistant, image, system, tool, tool_result, user
+from xai_sdk.tools import code_execution, get_tool_call_type, mcp, web_search  # x_search not yet supported
+
+from .._run_context import RunContext
+from .._utils import now_utc
+from ..builtin_tools import CodeExecutionTool, MCPServerTool, WebSearchTool
+from ..exceptions import UserError
+from ..messages import (
+    BinaryContent,
+    BuiltinToolCallPart,
+    BuiltinToolReturnPart,
+    FinishReason,
+    ImageUrl,
+    ModelMessage,
+    ModelRequest,
+    ModelRequestPart,
+    ModelResponse,
+    ModelResponsePart,
+    ModelResponseStreamEvent,
+    SystemPromptPart,
+    TextPart,
+    ThinkingPart,
+    ToolCallPart,
+    ToolReturnPart,
+    UserPromptPart,
+)
+from ..models import (
+    Model,
+    ModelRequestParameters,
+    StreamedResponse,
+)
+from ..settings import ModelSettings
+from ..usage import RequestUsage
+
+
+class GrokModel(Model):
+    """A model that uses the xAI SDK to interact with Grok."""
+
+    _model_name: str
+    _api_key: str
+    _client: AsyncClient | None
+
+    def __init__(
+        self,
+        model_name: str,
```

Comment:
As with `OpenAIModel` etc, can we use a type that has the allowed values so it can auto-complete in the IDE?